### DISCUSSION TITLE
celery 5.2.2 AttributeError: 'Settings' object has no attribute 'worker_pool'

### DISCUSSION THREAD

**Discussion by chenyunhsin** [celery/celery#8089] [Q&A]
# Checklist

- [ x] I have verified that the issue exists against the `main` branch of Celery.
- [x ] This has already been asked to the [discussions forum](https://github.com/celery/celery/discussions) first.
- [ x] I have read the relevant section in the
  [contribution guide](https://docs.celeryq.dev/en/main/contributing.html#other-bugs)
  on reporting bugs.
- [x ] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Bug+Report%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical bug reports.
- [ x] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22PR+Type%3A+Bugfix%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
- [x ] I have checked the [commit log](https://github.com/celery/celery/commits/main)
  to find out if the bug was already fixed in the main branch.
- [ x] I have included all related issues and possible duplicate issues
  in this issue (If there are none, check this box anyway).

## Mandatory Debugging Information
Hi all, I want to upgrade celery from 4.4.0 to 5.2.2, but when I set --pool parameters, I always trigger this error with no ideas.
**AttributeError: 'Settings' object has no attribute 'worker_pool'**

my command
`service celeryd start` and it will call celeryd,the celeryd file is a copy from [here](https://github.com/celery/celery/commits/main/extra/generic-init.d/celeryd), 
I only add some parameters when celeryd starts shows key error, the part I added:

<img width="537" alt="截圖 2023-02-18 上午9 48 32" src="https://user-images.githubusercontent.com/58776036/219825254-41d9e6aa-36a7-4812-b0ac-e3e94601d717.png">

it seems work well when I encounter  **thAttributeError: 'Settings' object has no attribute 'xxx'** of these two parameters **--statedb="cloud_celery_db" \  --prefetch-multiplier=4** 
but not **worker_pool**
my confuse is the terminal told me I loss parameter worker_pool(**AttributeError: 'Settings' object has no attribute 'worker_pool'**),  
so I add --pool="prefork", 
it still came with same error, 
but if I use --pool "prefork"(`I found someone did this on stack overflow`), it will said that pool parameter error(`it seems get the parameter behind --pool, like --statedb="cloud_celery_db"`)
 I have been confused with this parameter for a whole day, someone who can help I will be very appreciated !! 
how to set worker_pool!!

- [x ] I have included the output of ``celery -A proj report`` in the issue.
    (if you are not able to do this, then at least specify the Celery
     version affected).
- [x ] I have verified that the issue exists against the `main` branch of Celery.
- [ x] I have included the contents of ``pip freeze`` in the issue.
- [ x] I have included all the versions of all the external dependencies required
  to reproduce this bug.

## Optional Debugging Information

- [ x] I have tried reproducing the issue on more than one Python version
  and/or implementation.
- [x ] I have tried reproducing the issue on more than one message broker and/or
  result backend.
- [ x] I have tried reproducing the issue on more than one version of the message
  broker and/or result backend.
- [ x] I have tried reproducing the issue on more than one operating system.
- [x ] I have tried reproducing the issue on more than one workers pool.
- [ x] I have tried reproducing the issue with autoscaling, retries,
  ETA/Countdown & rate limits disabled.
- [x ] I have tried reproducing the issue after downgrading
  and/or upgrading Celery and its dependencies.

#### Related Issues

- None

#### Possible Duplicates

- None

## Environment & Settings

**Celery version**: 5.2.2

<img width="860" alt="截圖 2023-02-17 下午2 36 49" src="https://user-images.githubusercontent.com/58776036/219567891-31f9c9fe-10a5-4a3a-9545-6f3d6f51f2a1.png">

<details>
<summary><b><code>celery report</code> Output: </b>
<img width="962" alt="截圖 2023-02-17 下午2 42 27" src="https://user-images.githubusercontent.com/58776036/219569173-a7834f8b-7264-4d9e-a245-8c3b473ffd6d.png">
</summary>
<p>

</p>
</details>

# Steps to Reproduce
download celery 5.2.2 and try my command?
## Required Dependencies

- **Minimal Python Version**: 3.7.12
- **Minimal Celery Version**: 5.2.2
- **Minimal Kombu Version**:  Unknown
- **Minimal Broker Version**: N/A
- **Minimal Result Backend Version**: N/A 
- **Minimal OS and/or Kernel Version**: ubuntu 18.04
- **Minimal Broker Client Version**: N/A
- **Minimal Result Backend Client Version**: N/A

### Python Packages

<details>
<summary><b><code>pip freeze</code> Output:
adal==1.2.7; python_full_version >= "3.6.0"
amqp==5.1.1; python_version >= "3.7"
antlr4-python3-runtime==4.7.2; python_full_version >= "3.6.0"
applicationinsights==0.11.10; python_full_version >= "3.6.0"
argcomplete==1.12.3; python_full_version >= "3.6.0"
async-timeout==4.0.2; python_version >= "3.7"
azure-appconfiguration==1.1.1; python_full_version >= "3.6.0"
azure-batch==11.0.0; python_full_version >= "3.6.0"
azure-cli-core==2.30.0; python_full_version >= "3.6.0"
azure-cli-telemetry==1.0.6
azure-cli==2.30.0; python_full_version >= "3.6.0"
azure-common==1.1.28; python_version >= "3.7" and python_full_version >= "3.6.0"
azure-core==1.26.3; python_version >= "3.7" and python_full_version >= "3.6.0"
azure-cosmos==3.2.0; python_full_version >= "3.6.0"
azure-datalake-store==0.0.52; python_full_version >= "3.6.0"
azure-graphrbac==0.60.0; python_full_version >= "3.6.0"
azure-identity==1.12.0; python_version >= "3.7" and python_full_version >= "3.6.0"
azure-keyvault-administration==4.0.0b3; python_full_version >= "3.6.0"
azure-keyvault-keys==4.5.1; python_version >= "3.6" and python_full_version >= "3.6.0"
azure-keyvault-secrets==4.2.0
azure-keyvault==1.1.0; python_full_version >= "3.6.0"
azure-loganalytics==0.1.1; python_full_version >= "3.6.0"
azure-mgmt-advisor==9.0.0; python_full_version >= "3.6.0"
azure-mgmt-apimanagement==0.2.0; python_full_version >= "3.6.0"
azure-mgmt-appconfiguration==2.0.0; python_full_version >= "3.6.0"
azure-mgmt-applicationinsights==1.0.0; python_full_version >= "3.6.0"
azure-mgmt-authorization==0.61.0; python_full_version >= "3.6.0"
azure-mgmt-batch==16.0.0; python_full_version >= "3.6.0"
azure-mgmt-batchai==7.0.0b1; python_full_version >= "3.6.0"
azure-mgmt-billing==6.0.0; python_full_version >= "3.6.0"
azure-mgmt-botservice==0.3.0; python_full_version >= "3.6.0"
azure-mgmt-cdn==11.0.0; python_full_version >= "3.6.0"
azure-mgmt-cognitiveservices==12.0.0; python_full_version >= "3.6.0"
azure-mgmt-compute==23.1.0; python_full_version >= "3.6.0"
azure-mgmt-consumption==2.0.0; python_full_version >= "3.6.0"
azure-mgmt-containerinstance==9.1.0; python_full_version >= "3.6.0"
azure-mgmt-containerregistry==8.2.0; python_full_version >= "3.6.0"
azure-mgmt-containerservice==16.1.0; python_full_version >= "3.6.0"
azure-mgmt-core==1.3.2; python_version >= "3.7" and python_full_version >= "3.6.0"
azure-mgmt-cosmosdb==7.0.0; python_version >= "3.6" and python_full_version >= "3.6.0"
azure-mgmt-databoxedge==1.0.0; python_full_version >= "3.6.0"
azure-mgmt-datalake-analytics==0.2.1; python_full_version >= "3.6.0"
azure-mgmt-datalake-nspkg==3.0.1; python_full_version >= "3.6.0"
azure-mgmt-datalake-store==0.5.0; python_full_version >= "3.6.0"
azure-mgmt-datamigration==10.0.0; python_full_version >= "3.6.0"
azure-mgmt-deploymentmanager==0.2.0; python_full_version >= "3.6.0"
azure-mgmt-devtestlabs==4.0.0; python_full_version >= "3.6.0"
azure-mgmt-dns==8.0.0; python_full_version >= "3.6.0"
azure-mgmt-eventgrid==9.0.0; python_full_version >= "3.6.0"
azure-mgmt-eventhub==9.1.0; python_full_version >= "3.6.0"
azure-mgmt-extendedlocation==1.0.0; python_full_version >= "3.6.0"
azure-mgmt-hdinsight==9.0.0; python_full_version >= "3.6.0"
azure-mgmt-imagebuilder==0.4.0; python_full_version >= "3.6.0"
azure-mgmt-iotcentral==9.0.0; python_full_version >= "3.6.0"
azure-mgmt-iothub==2.1.0; python_full_version >= "3.6.0"
azure-mgmt-iothubprovisioningservices==0.3.0; python_full_version >= "3.6.0"
azure-mgmt-keyvault==9.2.0; python_full_version >= "3.6.0"
azure-mgmt-kusto==0.3.0; python_full_version >= "3.6.0"
azure-mgmt-loganalytics==11.0.0; python_full_version >= "3.6.0"
azure-mgmt-managedservices==1.0.0; python_full_version >= "3.6.0"
azure-mgmt-managementgroups==0.2.0; python_full_version >= "3.6.0"
azure-mgmt-maps==2.0.0; python_full_version >= "3.6.0"
azure-mgmt-marketplaceordering==1.1.0; python_full_version >= "3.6.0"
azure-mgmt-media==7.0.0; python_full_version >= "3.6.0"
azure-mgmt-monitor==2.0.0; python_full_version >= "3.6.0"
azure-mgmt-msi==0.2.0; python_full_version >= "3.6.0"
azure-mgmt-netapp==5.1.0; python_full_version >= "3.6.0"
azure-mgmt-network==19.1.0; python_full_version >= "3.6.0"
azure-mgmt-nspkg==3.0.2; python_full_version >= "3.6.0"
azure-mgmt-policyinsights==1.0.0; python_full_version >= "3.6.0"
azure-mgmt-privatedns==1.0.0; python_full_version >= "3.6.0"
azure-mgmt-rdbms==10.0.0; python_full_version >= "3.6.0"
azure-mgmt-recoveryservices==2.0.0; python_full_version >= "3.6.0"
azure-mgmt-recoveryservicesbackup==3.0.0; python_full_version >= "3.6.0"
azure-mgmt-redhatopenshift==1.0.0; python_full_version >= "3.6.0"
azure-mgmt-redis==13.0.0; python_full_version >= "3.6.0"
azure-mgmt-relay==0.1.0; python_full_version >= "3.6.0"
azure-mgmt-reservations==0.6.0; python_full_version >= "3.6.0"
azure-mgmt-resource==19.0.0; python_full_version >= "3.6.0"
azure-mgmt-search==8.0.0; python_full_version >= "3.6.0"
azure-mgmt-security==2.0.0; python_version >= "3.7" and python_full_version >= "3.6.0"
azure-mgmt-servicebus==6.0.0; python_full_version >= "3.6.0"
azure-mgmt-servicefabric==1.0.0; python_full_version >= "3.6.0"
azure-mgmt-servicefabricmanagedclusters==1.0.0; python_full_version >= "3.6.0"
azure-mgmt-servicelinker==1.0.0b1; python_full_version >= "3.6.0"
azure-mgmt-signalr==1.0.0; python_full_version >= "3.6.0"
azure-mgmt-sql==3.0.1; python_full_version >= "3.6.0"
azure-mgmt-sqlvirtualmachine==1.0.0b5; python_version >= "3.7" and python_full_version >= "3.6.0"
azure-mgmt-storage==19.0.0; python_full_version >= "3.6.0"
azure-mgmt-synapse==2.1.0b7; python_version >= "3.7" and python_full_version >= "3.6.0"
azure-mgmt-trafficmanager==0.51.0; python_full_version >= "3.6.0"
azure-mgmt-web==4.0.0; python_full_version >= "3.6.0"
azure-multiapi-storage==0.7.0; python_full_version >= "3.6.0"
azure-nspkg==3.0.2; python_full_version >= "3.6.0"
azure-storage-common==1.4.2; python_full_version >= "3.6.0"
azure-synapse-accesscontrol==0.5.0; python_full_version >= "3.6.0"
azure-synapse-artifacts==0.8.0; python_full_version >= "3.6.0"
azure-synapse-managedprivateendpoints==0.3.0; python_full_version >= "3.6.0"
azure-synapse-spark==0.2.0; python_full_version >= "3.6.0"
bcrypt==4.0.1; python_version >= "3.6" and python_full_version >= "3.6.0"
billiard==3.6.4.0; python_version >= "3.7"
cached-property==1.5.2; python_version < "3.8" and python_version >= "3.7"
celery==5.2.2; python_version >= "3.7"
certifi==2022.12.7; python_version >= "3.7" and python_full_version >= "3.6.0"
cffi==1.15.1; python_version >= "3.7" and python_full_version >= "3.6.0"
chardet==3.0.4; python_full_version >= "3.6.0" and python_version >= "3.7"
click-didyoumean==0.3.0; python_full_version >= "3.6.2" and python_full_version < "4.0.0" and python_version >= "3.7"
click-plugins==1.1.1; python_version >= "3.7"
click-repl==0.2.0; python_version >= "3.7"
click==8.1.3; python_full_version >= "3.6.2" and python_full_version < "4.0.0" and python_version >= "3.7"
colorama==0.4.6; platform_system == "Windows" and python_version >= "3.7" and python_full_version >= "3.7.0"
cryptography==39.0.1
deprecated==1.2.13; python_version >= "3.7" and python_full_version >= "3.6.0"
distro==1.8.0; python_version >= "3.6" and python_full_version >= "3.6.0"
django-celery-beat==2.1.0
django-celery-results==2.0.0
django-cors-headers==3.1.1; python_version >= "3.5"
django-mongoengine==0.4.1; python_version >= "2.7"
django-redis==4.12.1; python_version >= "3.5"
django-timezone-field==4.2.3; python_version >= "3.5"
django==2.2.28; python_version >= "3.5"
djangorestframework-simplejwt==4.6.0; python_version >= "3.7"
djangorestframework==3.12.2; python_version >= "3.5"
dnspython==2.3.0; python_version >= "3.7" and python_version < "4.0"
docker==5.0.3; python_version >= "3.6"
fabric==2.7.1; python_full_version >= "3.6.0"
humanfriendly==9.2; python_full_version >= "3.6.0"
idna==2.10; python_full_version >= "3.6.0" and python_version >= "3.7"
importlib-metadata==4.2.0; python_version == "3.7" and python_full_version >= "3.6.0"
invoke==1.7.3; python_full_version >= "3.6.0"
isodate==0.6.1; python_version >= "3.7" and python_full_version >= "3.6.0"
javaproperties==0.5.2; python_full_version >= "3.6.0" and python_version < "4"
jmespath==1.0.1; python_version >= "3.7" and python_full_version >= "3.6.0"
jsondiff==1.2.0; python_full_version >= "3.6.0"
knack==0.8.2; python_full_version >= "3.6.0"
kombu==5.2.4; python_version >= "3.7"
mongoengine==0.24.1; python_version >= "3.6"
msal-extensions==0.3.1; python_version >= "3.7" and python_full_version >= "3.6.0"
msal==1.21.0; python_version >= "3.7" and python_full_version >= "3.6.0"
msrest==0.7.1; python_version >= "3.7" and python_full_version >= "3.6.0"
msrestazure==0.6.4; python_full_version >= "3.6.0"
mysqlclient==2.0.1; python_version >= "3.5"
oauthlib==3.2.2; python_version >= "3.7" and python_full_version >= "3.6.0"
packaging==20.9; python_full_version >= "3.6.0"
paramiko==2.12.0; python_full_version >= "3.6.0"
pathlib2==2.3.7.post1; python_full_version >= "3.6.0"
pkginfo==1.9.6; python_version >= "3.6" and python_full_version >= "3.6.0"
portalocker==1.7.1
prompt-toolkit==3.0.36; python_full_version >= "3.6.2" and python_version >= "3.7"
psutil==5.9.4; python_full_version >= "3.6.0"
pycparser==2.21; python_version >= "3.7" and python_full_version >= "3.6.0"
pycryptodome==3.17; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.5.0")
pygithub==1.57; python_version >= "3.7" and python_full_version >= "3.6.0"
pygments==2.14.0; python_version >= "3.6" and python_full_version >= "3.6.0"
pyjwt==2.6.0; python_version >= "3.7"
pymongo==4.3.3; python_version >= "3.7"
pynacl==1.4.0; python_version >= "3.7" and python_full_version >= "3.6.0"
pyopenssl==23.0.0; python_version >= "3.7" and python_full_version >= "3.6.0"
pyparsing==3.0.9; python_full_version >= "3.6.8"
pyreadline==2.1; sys_platform == "win32" and python_full_version >= "3.6.0"
pysocks==1.7.1; python_full_version >= "3.6.0" and python_version >= "3.7"
python-crontab==2.7.1
python-dateutil==2.8.2; python_full_version >= "3.6.0"
python-dotenv==0.19.2; python_version >= "3.5"
pytz==2019.1; python_version >= "3.7" and python_full_version >= "3.6.0"
pywin32==227; sys_platform == "win32" and python_version >= "3.6" and platform_system == "Windows"
pyyaml==6.0; python_version >= "3.6" and python_full_version >= "3.6.0"
redis==4.5.1; python_version >= "3.7"
requests-oauthlib==1.3.1; python_version >= "3.7" and python_full_version >= "3.6.0"
requests==2.25.1; python_full_version >= "3.6.0" and python_version >= "3.7"
scp==0.13.6; python_full_version >= "3.6.0"
semver==2.13.0; python_full_version >= "3.6.0"
six==1.16.0; python_version >= "3.7" and python_full_version >= "3.6.0" and python_version < "4" and (python_version >= "3.7" and python_full_version < "3.0.0" or python_full_version >= "3.3.0" and python_version >= "3.7")
sqlparse==0.4.3; python_version >= "3.7"
sshtunnel==0.1.5; python_full_version >= "3.6.0"
tabulate==0.9.0; python_version >= "3.7" and python_full_version >= "3.6.0"
typing-extensions==4.5.0; python_version == "3.7" and python_full_version >= "3.6.0"
urllib3-secure-extra==0.1.0; python_full_version >= "3.6.0" and python_version >= "3.7"
urllib3==1.26.14; python_full_version >= "3.6.0" and python_version >= "3.7"
vine==5.0.0; python_version >= "3.7"
wcwidth==0.2.6; python_full_version >= "3.6.2" and python_version >= "3.7"
websocket-client==0.56.0; python_full_version >= "3.6.0" and python_version >= "3.6"
wrapt==1.14.1; python_version >= "3.7" and python_full_version >= "3.6.0"
xmltodict==0.13.0; python_version >= "3.4" and python_full_version >= "3.6.0"
zipp==3.13.0; python_version == "3.7" and python_full_version >= "3.6.0"
</b></summary>
<p>

```
```

</p>
</details>

### Other Dependencies

<details>
<p>
N/A
</p>
</details>

## Minimally Reproducible Test Case

<details>
<p>

```python
```

</p>
</details>

# Expected Behavior

# Actual Behavior

[a comment by an automated account is not shown]

**Comment by chenyunhsin** [comment]
since celery 5, celeryd is never used, done 0.0

**Comment by chenyunhsin** [comment]
I found it will need demonization in production environment, so it still need celeryd and celery beat files...So my problem still exist, anyone could help will be very appreciate QQ
https://docs.celeryq.dev/en/stable/userguide/daemonizing.html#daemonizing

**Comment by AlexanderNeilson** [comment]
I ran into what seems to be this same error in my system (still working through other problems).

Please check if you have any os.environ['x'] style loads in your settings file for django and instead try os.environ.get('x') and see if that works around the issue.
